File: Silver/1094/1094.js
// 막대기

const fs = require("fs");
/* 제출용 */
let input = fs.readFileSync('/dev/stdin').toString().trim();
/* 테스트용 */
// let input = fs.readFileSync(__dirname + '/input.txt').toString().trim();
let X = Number(input);
let bar = 64;
let count = 0;

while(true) {
  if (bar === X) {
    count++;
    break;
  }
  else if (bar > X) {
    bar /= 2;
  }else {
    count++;
    X -= bar;
  }
}

console.log(count);

File: Silver/1094/1094.py
# 막대기

x = int(input())

n = [64]
i = 0

while(1):
  if(sum(n) == x):
    break
  elif (sum(n) > x):
    n[i] = n[i] // 2
  else:
    n.append(n[i] // 2)
    i += 1

print(len(n))